### PR TITLE
Improving documentation

### PR DESCRIPTION
**Pull request by Tom32i**

### CODE REVIEW

File: doc/processors.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+Content processor are the main entry point for custom behaviours in Stenope.
+
+## What are processors
+
+Processors intervene in the content loading process, after the raw content is parsed and normalized into an array, and before it's hydrated into a model object.
+
+Writing a custom processor allow you to aply virtualiy any modification to the normalized content before it's returned as an object.
+
+Internaly, Stenope has processors for:
```
Comment (ogizanagi) [inline doc/processors.md:11]:
```suggestion
Internally, Stenope registers default processors for:
```

File: doc/processors.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Content processors
+
+Content processor are the main entry point for custom behaviours in Stenope.
+
+## What are processors
+
+Processors intervene in the content loading process, after the raw content is parsed and normalized into an array, and before it's hydrated into a model object.
```
Comment (ogizanagi) [inline doc/processors.md:7]: To shorten/rephrase:

```suggestion
A processor acts on the raw content loaded from your sources as an array, before denormalization into its model.
```

?

File: doc/processors.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+Writing a custom processor allow you to aply virtualiy any modification to the normalized content before it's returned as an object.
```
Comment (ogizanagi) [inline doc/processors.md:9]:
```suggestion
Writing a custom processor allow you to apply virtually any modification to the normalized content before denormalization.
```

File: doc/processors.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+Internaly, Stenope has processors for:
+- Highlighting code blocks with synthaxic coloration.
```
Comment (ogizanagi) [inline doc/processors.md:12]:
```suggestion
- Syntax highlighting of code blocks.
```

**Review by ogizanagi — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1,35 +1,73 @@
...
 ## Usage
 
-Build the static site: `bin/console -e prod stenope:build`
+Just [install Stenope](#installation) and run:
+
+    bin/console -e prod stenope:build ./static
+
+Your Symfony app is now a static website in: `./static`! ✨
+
+## Phylosophy and goals
```
Comment (ogizanagi) [inline README.md:33]:
```suggestion
## Philosophy and goals
```

File: README.md [hunk 6]
```diff
@@ -1,35 +1,73 @@
...
-- "You should adapt it to your need, don't adapt your needs to it".
-- "It connects with standard Symfony components and feels natural to Symfony developers".
+![CLI](cli.png)
 
-## How it works
+## Why another static site generator?
+
+We're Symfony developers that couldn't find a static website generator that bended to our needs.
+
+We needed a way of generatating static websites that:
```
Comment (ogizanagi) [inline README.md:11]:
```suggestion
We needed a way of generating static websites that:
```

gneu gneu gnaratating

File: README.md [hunk 7]
```diff
@@ -1,35 +1,73 @@
...
+
+- Doesn't impose a format, a structure or a location for the data.
+- Runs in a [Symfony](symfony.com) environement.
```
Comment (ogizanagi) [inline README.md:14]:
```suggestion
- Runs in a [Symfony](symfony.com) environment.
```

File: README.md [hunk 8]
```diff
@@ -1,35 +1,73 @@
...
+- Stenope meets your needs, not the other way around.
+- Stenope runs in any Symfony project out of the box, connects with standard Symfony components and feels natural to Symfony developers.
+- Stenope is highly extensible: features can be replaced, added or removed.
 
-## Advanced usage
+## How it works
+
+- 🔍 Stenope scans your Symfony app (like a search engine crawler would) and dumps every page into a static HTML file.
+- 🛠 Stenope provides tools for loading and parsing various data sources (like local Markdown files or distant headless CMS).
+- 🖌 Stenope enriches the parsed data by applying a serie of modificators (like Syntax Highlinting, slug generation, etc.).
```
Comment (ogizanagi) [inline README.md:45]:
```suggestion
- 🖌 Stenope enriches the parsed data by applying a series of processors (like Syntax Highlighting, slug generation, etc.).
```

File: README.md [hunk 9]
```diff
@@ -1,35 +1,73 @@
...
+- 🧲 Stenope finally hydrate your custom PHP objects with the enriched data and provides interfaces for listing and retrieving them (like an ORM would).
```
Comment (ogizanagi) [inline README.md:46]:
```suggestion
- 🧲 Stenope finally hydrates your custom PHP objects with the enriched data and provides interfaces for listing and retrieving them (like an ORM would).
```

File: README.md [hunk 10]
```diff
@@ -1,35 +1,73 @@
...
+- ⚙️ Stenope gives you a lot of control over the whole process by providing entry points, interfaces and default implementations that are entirely replaceable.
```
Comment (ogizanagi) [inline README.md:47]:
```suggestion
- ⚙️ Stenope gives you a lot of control over the whole process by providing entrypoints, interfaces and default implementations that are entirely replaceable.
```

File: doc/build-configuration.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
+# Configuring the build
+
+## Setting the build directory
+
+The build directory is the folder in which all the static content gets dumped.
+
+Their's several way to tell Stenope where to dump the static files:
```
Comment (ogizanagi) [inline doc/build-configuration.md:7]:
```suggestion
There are several way to tell Stenope where to dump the static files:
```

File: doc/build-configuration.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    build_dir: '%kernel.project_dir%/build'
+```
+
+### From command line argument
+
+```shell
+bin/console stenope:build ./path/to/the/build
+```
+
+## Controlling wich content gets in the build
```
Comment (ogizanagi) [inline doc/build-configuration.md:23]:
```suggestion
## Controlling which content gets in the build
```

File: doc/build-configuration.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    'stenope' => [
+        'ignore' => true,
+    ],
+])]
+public function fooAction() { /* ... */ }
+```
+
+### Excluding a route from the sitemap
+
+If you need to exclude some routes from the generated sitemap, but keep them in the build anyway, set the `stenope.sitemap` route option to `false`:
```
Comment (ogizanagi) [inline doc/build-configuration.md:44]:
```suggestion
If you need to exclude some routes from the generated sitemap, but still keep them in the build, set the `stenope.sitemap` route option to `false`:
```

File: doc/loading-content.md [hunk 14]
```diff
@@ -91,7 +93,9 @@ _Note: contents of the same type can very well be writen in different formats._
 
 ### Fetching a specific content
 
-The ContentManager uses slugs to identify your content. The `slug` argument must exactly matche the static file name in your content directory.
+The ContentManager uses slugs to identify your content.
+
+The `slug` argument must exactly matche the static file name in your content directory.
```
Comment (ogizanagi) [inline doc/loading-content.md:98]:
```suggestion
The `slug` argument must exactly match the static file name in your content directory.
```

File: doc/supported-sources.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
+# Supported sources
+
+## Embeded providers
+
+### Local file system
+
+The `LocalFilesystemProvider` is based on Symfony's Finder component and allow you to load contents from the local file system by attributing a path to a content type.
```
Comment (ogizanagi) [inline doc/supported-sources.md:7]:
```suggestion
The `LocalFilesystemProvider` is based on Symfony's Finder component and allows you to load contents from the local file system by attributing a path to a content type.
```

**Review by ogizanagi — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (Tom32i) [reply README.md:11]: ^^

**Pull request merged**